processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -298,38 +297,15 @@ else if ( reportErrorOnTargetObject( mapping ) ) {
 
         private void handleDefinedNestedTargetMapping(Set<String> handledTargets) {
 
-            Map<PropertyEntry, MappingOptions> optionsByNestedTarget =
-                method.getMappingOptions().groupByPoppedTargetReferences();
-            for ( Entry<PropertyEntry, MappingOptions> entryByTP : optionsByNestedTarget.entrySet() ) {
+            NestedTargetPropertyMappingHolder holder = new NestedTargetPropertyMappingHolder.Builder()
```

Comment:
That's not what I meant.. We do need the "old" routine.. But what is less clear is that there are 2 paths to get to the old routine.. One direct (target is not nested) and one after popping via nested.  (`method.getMappingOptions().hasNestedTargetReferences()` considers all mappings)..